src-electron/generator/template-util.js

```diff
@@ -142,23 +142,30 @@ async function ensureZclPackageId(context) {
  * @returns promise that resolves with a list of package id.
  */
 async function ensureZclPackageIds(context) {
-  // Get category of templates.json
-  let pkg = await queryPackage.getPackageByPackageId(
-    context.global.db,
-    context.global.genTemplatePackageId
-  )
-  let packageCategory = pkg ? pkg.category : null
+  let packageCategory = null
+  if (context.global.genTemplatePackage != null) {
+    packageCategory = context.global.genTemplatePackage.category
```

Comment:
So we do not have genTemplatePackageInfo in the context object already?
Wondering if this info already existed as well?

Reply:
We were just storing the IDs. I made the whole package object be stored in the cache. That help in cases where you need a category, since you have it already there, instead of having to look it up from the DB. That alone solved 25% of performance issues.

Reply:
Should we be doing `packageCategory = await ensureTemplatePackageCategory(context)` ?

Reply:
Yep.

Reply:
I meant instead of doing `packageCategory = context.global.genTemplatePackage.category`. Does that code need to be updated to `packageCategory = await ensureTemplatePackageCategory(context)`